### PR TITLE
Add more haptic feedback varieties

### PR DESCRIPTION
**Pull request by xster**
https://github.com/flutter/flutter/issues/13710

Primarily drove nomenclature off of iOS APIs since iOS is a bit more opinionated about haptic types. The Android consts are based off of Nexus 5X.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -76,8 +76,22 @@ - (void)playSystemSound:(NSString*)soundType {
   }
 }
 
-- (void)vibrateHapticFeedback {
-  AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
+- (void)vibrateHapticFeedback:(NSString*)feedbackType {
+  if (!feedbackType) {
+    AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: `return;`

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 2]
```diff
@@ -76,8 +76,22 @@ - (void)playSystemSound:(NSString*)soundType {
   }
 }
 
-- (void)vibrateHapticFeedback {
-  AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
+- (void)vibrateHapticFeedback:(NSString*)feedbackType {
+  if (!feedbackType) {
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: `if (!feedbackType || !@available(iOS 10, *))`

**Comment by cbracken** [comment]
lgtm modulo comments

lgtm

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm (hunk 1 above)
Comment (xster) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: done

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm (hunk 2 above)
Comment (xster) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: ya, I did that initially but on second thought, I think we wouldn't want to buzz the vibrator on every picker scroll and similar circumstances. On iOS, the haptic engine type feedbacks like on pull to refresh etc are just ignored on older devices.

**Pull request merged**